burnman/anisotropy.py

```diff
@@ -0,0 +1,471 @@
+        return Tik
+
+    def linear_compressibility(self, direction):
+        """
+        Computes the linear compressibility in a given direction 
+        relative to the stiffness tensor
+        """
+        direction = normalize(direction)
+        Sijkk = np.einsum('ijkk', self.full_compliance_tensor)
+        beta = Sijkk.dot(direction).dot(direction)
+        return beta
+    
+    def youngs_modulus(self, direction):
+        """
+        Computes the Youngs modulus in a given direction 
+        relative to the stiffness tensor
+        """
+        direction = normalize(direction)
+        Sijkl = self.full_compliance_tensor
+        S = Sijkl.dot(direction).dot(direction).dot(direction).dot(direction)
+        return 1./S
+
+    def shear_modulus(self, plane_normal, shear_direction):
+        """
+        Computes the shear modulus on a plane in a given 
+        shear direction relative to the stiffness tensor
+        """
+        plane_normal = normalize(plane_normal)
+        shear_direction = normalize(shear_direction)
+        
+        assert np.abs(plane_normal.dot(shear_direction)) < np.finfo(np.float).eps, 'plane_normal and shear_direction must be orthogonal'
+        Sijkl = self.full_compliance_tensor
+        G = Sijkl.dot(shear_direction).dot(plane_normal).dot(shear_direction).dot(plane_normal)
+        return 0.25/G
+    
+    def poissons_ratio(self,
+                       axial_direction,
+                       lateral_direction):
+        """
+        Computes the poisson ratio given loading and response 
+        directions relative to the stiffness tensor
+        """
+        
+        axial_direction = normalize(axial_direction)
+        lateral_direction = normalize(lateral_direction)
+        assert np.abs(axial_direction.dot(lateral_direction)) < np.finfo(np.float).eps, 'axial_direction and lateral_direction must be orthogonal'
+        
+        Sijkl = self.full_compliance_tensor
+        x = axial_direction
+        y = lateral_direction
+        nu = -(Sijkl.dot(y).dot(y).dot(x).dot(x) / 
+               Sijkl.dot(x).dot(x).dot(x).dot(x) )
+        return nu
+    
+    def wave_velocities(self, propagation_direction):
+        """
+        Computes the compressional wave velocity, and two 
+        shear wave velocities in a given propagation direction
+
+        Returns two lists, containing the wave speeds and 
+        directions of particle motion relative to the stiffness tensor
+        """
+        propagation_direction = normalize(propagation_direction)
+        
+        Tik = self.christoffel_tensor(propagation_direction)
+
+        eigenvalues, eigenvectors = np.linalg.eig(Tik)
+
+        idx = eigenvalues.argsort()[::-1]   
+        eigenvalues = np.real(eigenvalues[idx])        
+        eigenvectors = eigenvectors[:,idx]
+        velocities = np.sqrt(eigenvalues/self.rho)
+
+        return velocities, eigenvectors
+
+    def plot_velocities(self):
```

Comment:
Should we have plotting functions within `Mineral` classes? This seems like a new pattern.